src/Roslyn.Diagnostics.Analyzers/UnitTests/DeclarePublicAPIAnalyzerTests.cs

```diff
@@ -1,100 +1,126 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System.Collections.Generic;
 using System.IO;
+using System.Threading.Tasks;
 using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.CodeFixes;
-using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Testing;
 using Microsoft.CodeAnalysis.Testing;
+using Microsoft.CodeAnalysis.Testing.Verifiers;
+using Microsoft.CodeAnalysis.VisualBasic.Testing;
 using Test.Utilities;
 using Xunit;
 
 namespace Roslyn.Diagnostics.Analyzers.UnitTests
 {
-    public class DeclarePublicAPIAnalyzerTests : CodeFixTestBase
+    public class DeclarePublicAPIAnalyzerTests
     {
-        protected override CodeFixProvider GetCSharpCodeFixProvider()
+        private static DiagnosticResult GetAdditionalFileResultAt(int line, int column, string path, DiagnosticDescriptor descriptor, params object[] arguments)
         {
-            return new DeclarePublicAPIFix();
+            return new DiagnosticResult(descriptor)
+                .WithLocation(path, line, column)
+                .WithArguments(arguments);
         }
 
-        protected override CodeFixProvider GetBasicCodeFixProvider()
+        private static DiagnosticResult GetCSharpResultAt(int line, int column, DiagnosticDescriptor descriptor, params object[] arguments)
         {
-            return new DeclarePublicAPIFix();
+            return new DiagnosticResult(descriptor)
+                .WithLocation(line, column)
+                .WithArguments(arguments);
         }
 
-        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
+        private static DiagnosticResult GetBasicResultAt(int line, int column, DiagnosticDescriptor descriptor, params object[] arguments)
         {
-            return new DeclarePublicAPIAnalyzer();
+            return new DiagnosticResult(descriptor)
+                .WithLocation(line, column)
+                .WithArguments(arguments);
         }
 
-        protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer()
+        private async Task VerifyBasicAsync(string source, string shippedApiText, string unshippedApiText, params DiagnosticResult[] expected)
         {
-            return new DeclarePublicAPIAnalyzer();
+            var test = new VisualBasicCodeFixTest<DeclarePublicAPIAnalyzer, DeclarePublicAPIFix, XUnitVerifier>
```

Comment:
Would be good to eventually move this into the base type so it can be shared between different additional file analyzer unit tests.

Reply:
There is no more base type, but we can implement a different verifier type (the type that gets aliased to `VerifyCS` and `VerifyVB`) with different sets of overloads.